internal/pkg/server/fleet_integration_test.go

```diff
@@ -541,6 +543,82 @@ func Test_SmokeTest_Agent_Calls(t *testing.T) {
 	require.Falsef(t, ok, "expected response to have no errors attribute, errors are present: %+v", ackObj)
 }
 
+func EnrollAgent(enrollBody string, t *testing.T, ctx context.Context, srv *tserver) string {
+	req, err := http.NewRequestWithContext(ctx, "POST", srv.baseURL()+"/api/fleet/agents/enroll", strings.NewReader(enrollBody))
+	require.NoError(t, err)
+	req.Header.Set("Authorization", "ApiKey "+srv.enrollKey)
+	req.Header.Set("User-Agent", "elastic agent "+serverVersion)
+	req.Header.Set("Content-Type", "application/json")
+
+	cli := cleanhttp.DefaultClient()
+	res, err := cli.Do(req)
+	require.NoError(t, err)
+
+	p, _ := io.ReadAll(res.Body)
+	res.Body.Close()
+	var obj map[string]interface{}
+	err = json.Unmarshal(p, &obj)
+	require.NoError(t, err)
+
+	t.Log(obj)
+
+	require.Equal(t, http.StatusOK, res.StatusCode)
+
+	item := obj["item"]
+	mm, ok := item.(map[string]interface{})
+	require.True(t, ok, "expected attribute item to be an object")
+	agentID := mm["id"]
+	return agentID.(string)
+}
+
+func Test_Agent_Enrollment_Id(t *testing.T) {
+	enrollBodyWEnrollmentID := `{
+	    "type": "PERMANENT",
+	    "shared_id": "",
+	    "enrollment_id": "123456",
+	    "metadata": {
+		"user_provided": {},
+		"local": {},
+		"tags": []
+	    }
+	}`
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Start test server
+	srv, err := startTestServer(t, ctx)
+	require.NoError(t, err)
+
+	t.Log("Enroll the first agent with enrollment_id")
+	firstAgentID := EnrollAgent(enrollBodyWEnrollmentID, t, ctx, srv)
+
+	t.Log("Enroll the second agent with the same enrollment_id")
+	secondAgentID := EnrollAgent(enrollBodyWEnrollmentID, t, ctx, srv)
```

Comment:
Could you write a version of this test where the API key of the agent being re-enrolled was invalidated separately? What if someone uses an independent query to revoke all API keys as a security measure? Could that break this implementation?

Reply:
Added one more test where the first api key was invalidated before the second enroll